### PR TITLE
Add `uninline-portable` feature

### PR DESCRIPTION
**Pull request by jarys**
Adds `uninline-portable` feature flag, which disables inlining of some functions. This is useful for tiny microchips (such as ARM Cortex-M0), where inlining can hurt performance and blow up binary size.

Inspired by [blake2b_simd/uninline_portable](https://github.com/oconnor663/blake2_simd/blob/master/blake2b/Cargo.toml#L16-L20) feature.

Saves 3 kb of space on Trezor T, where we definitely want to use this crate.

### CODE REVIEW

**Review by daira — APPROVED** [review]
utACK. I have not checked the generated code to see whether it is actually inlined or not depending on this feature.

File: src/fields/fp.rs [hunk 1]
```diff
@@ -293,7 +293,7 @@ impl Fp {
     }
 
     /// Squares this element.
-    #[inline]
+    #[cfg_attr(inline, not(feature = "uninline-portable"))]
```
Comment (str4d) [inline src/fields/fp.rs:296]: These are all backwards; `inline` is being used as the condition predicate, and `not(feature = "uninline-portable")` the applied attribute. I am surprised it compiles at all, let alone runs with the desired effect and passes tests.

Comment (jarys) [reply src/fields/fp.rs:296]: facepalm . But I'm not surprised it decreased size of the binary, because it definitely removes the `inline` attribute.

Comment (jarys) [reply src/fields/fp.rs]: fixed: 52af8293a8e31d27ae0cb712505cf6daf2f36549

**Comment by jarys** [comment]
Tested again. This saves 3584 bytes on Trezor T.

**Review by str4d — APPROVED** [review]
utACK

**Review by ebfull — APPROVED** [review]

**Pull request merged**